### PR TITLE
ENH Add loading of ORM field labels to i18nTextCollector

### PR DESCRIPTION
**Pull request by xini**
Replaces https://github.com/silverstripe/silverstripe-framework/pull/10686

### CODE REVIEW

File: src/i18n/TextCollection/i18nTextCollector.php [hunk 1]
```diff
@@ -920,6 +926,49 @@ public function collectFromEntityProviders($filePath, Module $module = null)
...
+     * @param string $filePath
+     * @return array
+     */
+    public function collectFromORM($filePath)
+    {
+        $entities = [];
+        $classes = ClassInfo::classes_for_file($filePath);
+        foreach ($classes as $class) {
+            // Skip non-implementing classes
+            if (!class_exists($class ?? '')) {
```
Comment (michalkleiner) [inline src/i18n/TextCollection/i18nTextCollector.php]: Is the null coalesce needed here? Can it be anything other than a string provided from `ClassInfo::classes_for_file`?

Comment (xini) [reply src/i18n/TextCollection/i18nTextCollector.php]: removed

**Comment by GuySartorelli** [comment]
As @michalkleiner requested in https://github.com/silverstripe/silverstripe-framework/pull/10686#issuecomment-1427334068:
> Can we update an existing test or add a new one for this new feature?

File: tests/php/i18n/i18nTextCollectorTest.php [hunk 2]
```diff
@@ -709,6 +712,43 @@ public function testCollectFromEntityProvidersInWebRoot()
...
+        ]);
+        Config::inst()->update(TestDataObject::class, 'has_one', [
+            'HasOneRelation' => Member::class
+        ]);
+        Config::inst()->update(TestDataObject::class, 'has_many', [
+            'HasManyRelation' => Member::class
+        ]);
+        Config::inst()->update(TestDataObject::class, 'many_many', [
+            'ManyManyRelation' => Member::class
+        ]);
```
Comment (xini) [inline tests/php/i18n/i18nTextCollectorTest.php]: @GuySartorelli I have added a test for this, but as you can see I had to update the config for the class manually. For whatever reason the config was not loaded from the class statics. Any idea why? Are the class names/namespaces in the tests handled differently?

Comment (xini) [reply tests/php/i18n/i18nTextCollectorTest.php]: bump

**Comment by GuySartorelli** [comment]
Sorry, I haven't had time to look at this again - at the very least this will need to target the `5` branch and be rebased. Can you please do that? I'll try make time to look at this properly over the next couple of weeks.

**Comment by xini** [comment]
Thanks @GuySartorelli. This is now rebased and squashed. The above config issue remains though.

**Review by GuySartorelli — APPROVED** [review]
Works great locally. Thanks for this!

**Comment by GuySartorelli** [comment]
https://github.com/silverstripe/developer-docs/pull/263 will document this change.

**Pull request merged**